tests/integration/profile_test.py

```diff
@@ -0,0 +1,142 @@
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+import os
+
+
+import pytest
+
+import libnmstate
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import InterfaceState
+
+from .testlib import cmd as libcmd
+
+
+DUMMY0_IFNAME = 'dummy0'
+
+NMCLI_CON_ADD_DUMMY_CMD = [
+    'nmcli',
+    'con',
+    'add',
+    'type',
+    'dummy',
+    'con-name',
+    'testProfile',
+    'connection.autoconnect',
+    'no',
+    'ifname',
+    DUMMY0_IFNAME,
+]
+
+NMCLI_CON_ADD_ETH_CMD = [
+    'nmcli',
+    'con',
+    'add',
+    'type',
+    'ethernet',
+    'con-name',
+    'testProfile',
+    'connection.autoconnect',
+    'no',
+    'ifname',
+    'eth1',
+]
+
+DUMMY_PROFILE_DIRECTORY = '/etc/NetworkManager/system-connections/'
+
+ETH_PROFILE_DIRECTORY = '/etc/sysconfig/network-scripts/'
+
+
+def test_delete_new_interface_inactive_profiles(dummy_inactive_profile):
+    with dummy_interface(dummy_inactive_profile):
+        profile_exists = _profile_exists(
+            DUMMY_PROFILE_DIRECTORY + 'testProfile.nmconnection'
+        )
+        assert not profile_exists
+
+
+def test_delete_existing_interface_inactive_profiles(eth1_up):
+    _create_inactive_profile('eth1')
+    eth1_up[Interface.KEY][0][Interface.MTU] = 2000
+    libnmstate.apply(eth1_up)
+    profile_exists = _profile_exists(
+        ETH_PROFILE_DIRECTORY + 'testProfile.nmconnection'
+    )
+    assert not profile_exists
+
+
+@contextmanager
+def dummy_interface(ifname):
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: ifname,
+                Interface.TYPE: InterfaceType.DUMMY,
+                Interface.STATE: InterfaceState.UP,
+            }
+        ]
+    }
+    libnmstate.apply(desired_state)
+    try:
+        yield
+    finally:
+        dummy0_dstate = desired_state[Interface.KEY][0]
+        dummy0_dstate[Interface.STATE] = InterfaceState.ABSENT
+        libnmstate.apply(desired_state)
+
+
+@pytest.fixture
+def dummy_inactive_profile():
+    libcmd.exec_cmd(NMCLI_CON_ADD_DUMMY_CMD)
+    profile_exists = _profile_exists(
+        DUMMY_PROFILE_DIRECTORY + 'testProfile.nmconnection'
+    )
+    assert profile_exists
+    try:
+        yield DUMMY0_IFNAME
+    finally:
+        libcmd.exec_cmd(_nmcli_delete_connection('testProfile'))
+
+
+def _create_inactive_profile(con_name):
```

Comment:
This duplicates the fixture code and is not marked as a contextmanager (and not used as such by the caller).

I do not think the deactivation is a must and after you make it a contextmanager, you can use it from the fixture.